### PR TITLE
Add diagnostics toggle

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

image
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/system/hassio-supervisor-info.ts [hunk 1]
```diff
@@ -17,7 +17,12 @@ import {
   setSupervisorOption,
   SupervisorOptions,
 } from "../../../src/data/hassio/supervisor";
-import { showConfirmationDialog } from "../../../src/dialogs/generic/show-dialog-box";
+import "../../../src/components/ha-switch";
+import {
+  showConfirmationDialog,
+  showAlertDialog,
+} from "../../../src/dialogs/generic/show-dialog-box";
+import "../../../src/panels/profile/ha-settings-row";
```
Comment (balloob) [inline hassio/src/system/hassio-supervisor-info.ts:25]: Let's move this to `src/components` now that it's used in multiple places.

File: hassio/src/system/hassio-supervisor-info.ts [hunk 2]
```diff
@@ -121,6 +144,13 @@ class HassioSupervisorInfo extends LitElement {
           color: var(--error-color);
           margin-top: 16px;
         }
+        ha-settings-row {
+          padding: 0;
+        }
+        .more {
```
Comment (balloob) [inline hassio/src/system/hassio-supervisor-info.ts:150]: Once it's a button:
```suggestion
        button {
```

File: hassio/src/system/hassio-supervisor-info.ts [hunk 3]
```diff
@@ -55,6 +60,23 @@ class HassioSupervisorInfo extends LitElement {
                 : ""}
             </tbody>
           </table>
+          <div class="options">
+            <ha-settings-row>
+              <span slot="heading">
+                Share Diagnostics
```
Comment (balloob) [inline hassio/src/system/hassio-supervisor-info.ts:66]: Let's make it clear we only share when things go wrong
```suggestion
                Share Crash Reports
```

Comment (balloob) [reply hassio/src/system/hassio-supervisor-info.ts:66]: Or is this so we can increase scope in the future without asking again?

Comment (pvizeli) [reply hassio/src/system/hassio-supervisor-info.ts:66]: Think this title "Share Diagnostics" is cleaner and we specify it on the next line with Crash Reports. I think it's more open in that way. In the end, we want only to see unknown issues which we can fix and no statistics or other things which are part of the updater on the core.

File: src/components/ha-settings-row.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  SVGTemplateResult,
+} from "lit-element";
+import "@polymer/paper-item/paper-item-body";
+
+@customElement("ha-settings-row")
+export class HaSettingsRow extends LitElement {
+  @property({ type: Boolean, reflect: true }) public narrow!: boolean;
+
+  @property({ type: Boolean, attribute: "three-line" })
+  public threeLine: boolean = false;
```
Comment (bramkragten) [inline src/components/ha-settings-row.ts:17]:
```suggestion
  public threeLine = false;
```

**Review by balloob — APPROVED** [review]

**Pull request merged**